src/chapters/errors-and-debugging/what-is-debugging.rst

```diff
@@ -0,0 +1,24 @@
+What is Debugging?
+==================
+
+.. index:: ! debugging, ! bug
+
+Programming is a complex process. Since it is done by human beings, errors often occur. Programming errors are called **bugs** and the process of tracking them down and correcting them is called **debugging**.  
+
+Some claim that in 1945, a dead moth caused a problem on relay number 70, panel F, of one of the first computers at Harvard, and the term bug has remained in use since. Wikipedia even has an image of the supposed `first bug <http://en.wikipedia.org/wiki/File:H96566k.jpg>`_.
+
+.. index::
+   single: error; syntax
+   single: error; runtime
+   single: error; logic
+
+Three kinds of errors can occur in a program: syntax errors, runtime errors, and logic errors. We will learn about each kind of error before discussing strategies for debugging code and reducing errors.
+
+Beginning Tips for Debugging
+----------------------------
+
+Debugging a program is a different way of thinking than writing a program. The process of debugging is much more like being a detective. Something has gone wrong and you must rely on clues, experience, intuition, and an inquisitive spirit to solve the problem.
+
+Everyone is a suspect (except JavaScript)!  It's common for beginner programmers to blame JavaScript, but that should be your last resort. Remember that JavaScript has been used to solve introductory coding problems millions of times by millions of other programmers. So, JavaScript is probably not the problem.
```

Comment:
Word removal: "Everyone is a suspect (except JavaScript)!  It's common for beginners to blame JavaScript. However, JavaScript has been used to solve coding problems millions of times by millions of users.  JavaScript is probably NOT the problem."

Reply:
I feel like this is a really polite way to say "you may be the problem when it comes to errors." I would recommend rewording it entirely to something more like: "Oftentimes, you will find it tempting to blame errors on Javascript itself. We encourage you to think critically about the code you have written and whether or not, you may have made an error in writing your code. Even senior developers drop a semi-colon once in a while!"

Reply:
Thanks for the suggestions. I cleaned this up quite a bit.